# Battle of the nieghborhoods

## INTRODUCTION/BUSINESS PROBLEM

Problem: Which city Pittsburgh or philadelphia would be a good place to start a coffee shop downtown?

I will look at the current venues and list the coffee shops that already exsist for competitor reasons.

Which city has more attractions to draw customers to the downtown area.

### Data I will be using

I will be using foursquare and looking for data to pull in and refine and use it to compare the too downtown areas

number of coffee shops within a center of each of the downtown cities

number and distance of coffee shops in the downtown

distance from center of the city

Following data sources will be needed to extract/generate the required information

centers of candidate areas will be generated algorithmically and approximate addresses of centers of those areas will be obtained using Google Maps API reverse geocoding
number of coffee shops and their type and location in every neighborhood will be obtained using Foursquare API
coordinate of Pittsburgh or Philadelphia center will be obtained using Google Maps API geocoding

In [1]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize

!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library

print('Folium installed')
print('Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    geopy-2.0.0                |     pyh9f0ad1d_0          63 KB  conda-forge
    ------------------------------------------------------------
                                           Total:          97 KB

The following NEW packages will be INSTALLED:

    geographiclib: 1.50-py_0          conda-forge
    geopy:         2.0.0-pyh9f0ad1d_0 conda-forge


geographiclib-1.50   | 34 KB     | ##################################### | 100% 
geopy-2.0.0          | 63 KB     | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Solving environment: done

# All requ

In [2]:
CLIENT_ID = 'KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1' # your Foursquare ID
CLIENT_SECRET = 'LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS' # your Foursquare Secret
VERSION = '20200701'
LIMIT = 30
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1
CLIENT_SECRET:LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS


In [3]:
address = '250 Forbes Avenue, Pittsburgh, PA, 15222, United States of America'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

40.44026183043749 -80.001540486808


In [4]:
search_query = 'coffee'
radius = 500
print(search_query + ' .... OK!')

coffee .... OK!


In [5]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&ll=40.44026183043749,-80.001540486808&v=20200701&query=coffee&radius=500&limit=30'

In [6]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5efc91cea50d067df950c74c'},
 'response': {'venues': [{'id': '4bd19d80046076b017157271',
    'name': '21st Street Coffee and Tea',
    'location': {'address': '225 5th Ave',
     'crossStreet': 'Btw Liberty Ave. & Wood St.',
     'lat': 40.44111204385811,
     'lng': -80.00099394922842,
     'labeledLatLngs': [{'label': 'display',
       'lat': 40.44111204385811,
       'lng': -80.00099394922842}],
     'distance': 105,
     'postalCode': '15222',
     'cc': 'US',
     'city': 'Pittsburgh',
     'state': 'PA',
     'country': 'United States',
     'formattedAddress': ['225 5th Ave (Btw Liberty Ave. & Wood St.)',
      'Pittsburgh, PA 15222',
      'United States']},
    'categories': [{'id': '4bf58dd8d48988d1e0931735',
      'name': 'Coffee Shop',
      'pluralName': 'Coffee Shops',
      'shortName': 'Coffee Shop',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_',
       'suffix': '.png'},
      'primary': True}],
 

In [7]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

categories delivery.id  \
0  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...         NaN   
1  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...     1526818   
2  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...         NaN   
3  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...         NaN   
4  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...         NaN   

               delivery.provider.icon.name  \
0                                      NaN   
1  /delivery_provider_grubhub_20180129.png   
2                                      NaN   
3                                      NaN   
4                                      NaN   

              delivery.provider.icon.prefix delivery.provider.icon.sizes  \
0                                       NaN                          NaN   
1  https://fastly.4sqi.net/img/general/cap/                     [40, 50]   
2                                       NaN                          NaN   
3                                       NaN                          NaN   
4                                       NaN                          NaN   

  delivery.provider.name                                       delivery.url  \
0                    NaN                                                NaN   
1                grubhub  https://www.grubhub.com/restaurant/nicholas-co...   
2                    NaN                                                NaN   
3                    NaN                                                NaN   
4                    NaN                                                NaN   

   hasPerk                        id location.address  ... location.distance  \
0    False  4bd19d80046076b017157271      225 5th Ave  ...               105   
1    False  4ad7b832f964a520070e21e3     23 Market Sq  ...               164   
2    False  4f32432419836c91c7c63ce0      239 4th Ave  ...                42   
3    False  56a406dc237cabfdf6483a1a     524 Penn Ave  ...               268   
4    False  58ee6754c960274628939615      1 Market St  ...               322   

                           location.formattedAddress  \
0  [225 5th Ave (Btw Liberty Ave. & Wood St.), Pi...   
1  [23 Market Sq, Pittsburgh, PA 15222, United St...   
2  [239 4th Ave, Pittsburgh, PA 15222, United Sta...   
3  [524 Penn Ave (Fifth Ave.), Pittsburgh, PA 152...   
4  [1 Market St, Pittsburgh, PA 15222, United Sta...   

                             location.labeledLatLngs location.lat  \
0  [{'label': 'display', 'lat': 40.44111204385811...    40.441112   
1  [{'label': 'display', 'lat': 40.44125, 'lng': ...    40.441250   
2  [{'label': 'display', 'lat': 40.439885, 'lng':...    40.439885   
3  [{'label': 'display', 'lat': 40.44241098442742...    40.442411   
4  [{'label': 'display', 'lat': 40.438336, 'lng':...    40.438336   

   location.lng      location.neighborhood location.postalCode  \
0    -80.000994                        NaN               15222   
1    -80.002990  Fifth and Forbes Corridor               15222   
2    -80.001465                        NaN               15222   
3    -80.002981                        NaN               15222   
4    -80.004385                        NaN               15222   

   location.state                           name    referralId  
0              PA     21st Street Coffee and Tea  v-1593611125  
1              PA  Nicholas Coffee & Tea Company  v-1593611125  
2              PA                    Coffee Shop  v-1593611125  
3              PA        Rock 'n' Joe Coffee Bar  v-1593611125  
4              PA               Fort Pitt Coffee  v-1593611125  

[5 rows x 24 columns]

In [8]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name         categories  \
0        21st Street Coffee and Tea        Coffee Shop   
1     Nicholas Coffee & Tea Company        Coffee Shop   
2                       Coffee Shop        Coffee Shop   
3           Rock 'n' Joe Coffee Bar        Coffee Shop   
4                  Fort Pitt Coffee        Coffee Shop   
5                   Coffee Ala Cart        Coffee Shop   
6     On19 Coffee Shop at K&L Gates        Coffee Shop   
7                 Cool Beans Coffee        Coffee Shop   
8                 Cool Beans Coffee        Coffee Shop   
9                       Coffee Shop      Deli / Bodega   
10         Creative Coffee + Supply        Coffee Shop   
11  Quick Stop Convenience & Coffee  Food & Drink Shop   
12                  Cart Coffee Ala               Food   
13           Tuscany Premium Coffee               Food   
14                   Redhawk Coffee        Coffee Shop   
15            The rockin coffee pot           Building   
16         Niko's Coffee Frick Bldg        Coffee Shop   
17                        Starbucks        Coffee Shop   
18        Kimpton Monaco Coffee Bar        Coffee Shop   
19            Lydiah's Coffee House        Coffee Shop   
20                        Starbucks        Coffee Shop   
21                        Starbucks        Coffee Shop   
22                      Crazy Mocha        Coffee Shop   
23                       Cool beans        Coffee Shop   
24            Gloria Jean's Coffees        Coffee Shop   

                                    address  cc        city        country  \
0                               225 5th Ave  US  Pittsburgh  United States   
1                              23 Market Sq  US  Pittsburgh  United States   
2                               239 4th Ave  US  Pittsburgh  United States   
3                              524 Penn Ave  US  Pittsburgh  United States   
4                               1 Market St  US  Pittsburgh  United States   
5                       525 William Penn Pl  US  Pittsburgh  United States   
6                           527-599 Wood St  US  Pittsburgh  United States   
7                              437 Grant St  US  Pittsburgh  United States   
8                           444 Liberty Ave  US  Pittsburgh  United States   
9                                       NaN  US         NaN  United States   
10                        309 Smithfield St  US  Pittsburgh  United States   
11                            Smithfield st  US  Pittsburgh  United States   
12                          444 Liberty Ave  US  Pittsburgh  United States   
13                              120 5th Ave  US  Pittsburgh  United States   
14  536 Smithfield St, Pittsburgh, PA 15222  US  Pittsburgh  United States   
15  67 Bedford Square, Pittsburgh, PA 15203  US  Pittsburgh  United States   
16                                      NaN  US  Pittsburgh  United States   
17                              7 Market Sq  US  Pittsburgh  United States   
18                  Level 1, Kimpton Monaco  US  Pittsburgh  United States   
19                                      NaN  US  Pittsburgh  United States   
20                            210 Sixth Ave  US  Pittsburgh  United States   
21                   530 William Penn Place  US  Pittsburgh  United States   
22                             500 Grant St  US  Pittsburgh  United States   
23                             707 Grant St  US  Pittsburgh  United States   
24                    201 Smithfield Street  US  Pittsburgh  United States   

                    crossStreet  distance  \
0   Btw Liberty Ave. & Wood St.       105   
1                           NaN       164   
2                           NaN        42   
3                    Fifth Ave.       268   
4                           NaN       322   
5                           NaN       339   
6                           6th       217   
7                           NaN       378   
8                           NaN       275   
9                           NaN       460  

In [9]:
dataframe_filtered.name

0          21st Street Coffee and Tea
1       Nicholas Coffee & Tea Company
2                         Coffee Shop
3             Rock 'n' Joe Coffee Bar
4                    Fort Pitt Coffee
5                     Coffee Ala Cart
6       On19 Coffee Shop at K&L Gates
7                   Cool Beans Coffee
8                   Cool Beans Coffee
9                         Coffee Shop
10           Creative Coffee + Supply
11    Quick Stop Convenience & Coffee
12                    Cart Coffee Ala
13             Tuscany Premium Coffee
14                     Redhawk Coffee
15              The rockin coffee pot
16           Niko's Coffee Frick Bldg
17                          Starbucks
18          Kimpton Monaco Coffee Bar
19              Lydiah's Coffee House
20                          Starbucks
21                          Starbucks
22                        Crazy Mocha
23                         Cool beans
24              Gloria Jean's Coffees
Name: name, dtype: object

In [10]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) # generate map centred around the Hilton Garden Inn, Pittsburgh

# add a red circle marker to represent the Hilton Garden Inn, Pittsburgh 
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Hilton Garden Inn, Pittsburgh',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)
# add the Italian restaurants as blue circle markers
for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.categories):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

explore closest coffe shop to hotel

In [11]:
venue_id = '4bd19d80046076b017157271' # ID of 21st Street Coffee and tea
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4bd19d80046076b017157271?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [12]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'popular', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4bd19d80046076b017157271',
 'name': '21st Street Coffee and Tea',
 'contact': {'phone': '4122810121',
  'formattedPhone': '(412) 281-0121',
  'twitter': '21streetcoffee'},
 'location': {'address': '225 5th Ave',
  'crossStreet': 'Btw Liberty Ave. & Wood St.',
  'lat': 40.44111204385811,
  'lng': -80.00099394922842,
  'labeledLatLngs': [{'label': 'display',
    'lat': 40.44111204385811,
    'lng': -80.00099394922842}],
  'postalCode': '15222',
  'cc': 'US',
  'city': 'Pittsburgh',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['225 5th Ave (Btw Liberty Ave. & Wood St.)',
   'Pittsburgh, PA 15222',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/21st-street-coffee-and-tea/4bd19d80046076b017157271',
 'categories': [{'id': '4bf58dd8d48988d1e0931735',
   'name': 'Coffee Shop',
   'pluralName': 'Coffee Shops',
   'shortName': 'Coffee Shop',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_',
    'suffix': '.png'},
 

In [13]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

6.6


In [14]:
venue_id = '4c336b3316adc9289395c39c' # ID of crazy mocha
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4c336b3316adc9289395c39c?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [15]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'popular', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'parent', 'hierarchy', 'attributes', 'bestPhoto'])


{'id': '4c336b3316adc9289395c39c',
 'name': 'Crazy Mocha',
 'contact': {'phone': '4122811981', 'formattedPhone': '(412) 281-1981'},
 'location': {'address': '500 Grant St',
  'lat': 40.44033967364289,
  'lng': -79.99530880229463,
  'labeledLatLngs': [{'label': 'display',
    'lat': 40.44033967364289,
    'lng': -79.99530880229463}],
  'postalCode': '15219',
  'cc': 'US',
  'city': 'Pittsburgh',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['500 Grant St',
   'Pittsburgh, PA 15219',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/crazy-mocha/4c336b3316adc9289395c39c',
 'categories': [{'id': '4bf58dd8d48988d1e0931735',
   'name': 'Coffee Shop',
   'pluralName': 'Coffee Shops',
   'shortName': 'Coffee Shop',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_',
    'suffix': '.png'},
   'primary': True}],
 'verified': False,
 'stats': {'tipCount': 8},
 'url': 'https://crazymocha.com',
 'likes': {'count': 8,
  'groups': [{

In [16]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

6.2


In [17]:
result['response']['venue']['tips']['count']

8

In [22]:
## Ecco Tips
limit = 15 # set limit to be greater than or equal to the total number of tips
url = 'https://api.foursquare.com/v2/venues/{}/tips?client_id={}&client_secret={}&v={}&limit={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION, limit)

results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5efc9b79d2bf83199c5b7771'},
 'response': {'tips': {'count': 8,
   'items': [{'id': '4d8b7cfdbc848cfab603f22b',
     'createdAt': 1300987133,
     'text': "Come for a laugh & ignore the haters. Try a white mocha w/ vanilla soy and you won't be disappointed!",
     'type': 'user',
     'canonicalUrl': 'https://foursquare.com/item/4d8b7cfdbc848cfab603f22b',
     'lang': 'en',
     'likes': {'count': 0, 'groups': []},
     'logView': True,
     'agreeCount': 0,
     'disagreeCount': 0,
     'todo': {'count': 0},
     'user': {'id': '1152389',
      'firstName': 'Marcus',
      'lastName': 'J',
      'photo': {'prefix': 'https://fastly.4sqi.net/img/user/',
       'suffix': '/1152389_5yIwGZPG_HUfcuaM8SYWscr9sNxK0fSicM6JTDlCGmvOhPYKVgn0rZ7KeldnA7ZMD1WFxOmwn.jpg'}}}]}}}

In [23]:
tips = results['response']['tips']['items']

tip = results['response']['tips']['items'][0]
tip.keys()

dict_keys(['id', 'createdAt', 'text', 'type', 'canonicalUrl', 'lang', 'likes', 'logView', 'agreeCount', 'disagreeCount', 'todo', 'user'])

In [24]:
pd.set_option('display.max_colwidth', -1)

tips_df = json_normalize(tips) # json normalize tips

# columns to keep
filtered_columns = ['text', 'agreeCount', 'disagreeCount', 'id', 'user.firstName', 'user.lastName', 'user.gender', 'user.id']
tips_filtered = tips_df.loc[:, filtered_columns]

# display tips
tips_filtered

/opt/conda/envs/Python36/lib/python3.6/site-packages/pandas/core/indexing.py:1494: FutureWarning: 
Passing list-likes to .loc or [] with any missing label will raise
KeyError in the future, you can use .reindex() as an alternative.

See the documentation here:
https://pandas.pydata.org/pandas-docs/stable/indexing.html#deprecate-loc-reindex-listlike
  return self._getitem_tuple(key)


text  \
0  Come for a laugh & ignore the haters. Try a white mocha w/ vanilla soy and you won't be disappointed!   

   agreeCount  disagreeCount                        id user.firstName  \
0  0           0              4d8b7cfdbc848cfab603f22b  Marcus          

  user.lastName  user.gender  user.id  
0  J            NaN           1152389

In [ ]:
user_id = '1552389' # user ID with most agree counts and complete profile

url = 'https://api.foursquare.com/v2/user/{}?client_id={}&client_secret={}&v={}'.format(user_id, CLIENT_ID, CLIENT_SECRET, VERSION) # define URL

# send GET request
results = requests.get(url).json()
user_data = results['response']['user']

# display features associated with user
user_data.keys()

In [ ]:
print('First Name: ' + user_data['firstName'])
print('Last Name: ' + user_data['lastName'])
print('Home City: ' + user_data['homeCity'])

In [ ]:
user_data['tips']

In [37]:
# define tips URL
url = 'https://api.foursquare.com/v2/users/{}/tips?client_id={}&client_secret={}&v={}&limit={}'.format(user_id, CLIENT_ID, CLIENT_SECRET, VERSION, limit)

# send GET request and get user's tips
results = requests.get(url).json()
tips = results['response']['tips']['items']

# format column width
pd.set_option('display.max_colwidth', -1)

tips_df = json_normalize(tips)

# filter columns
filtered_columns = ['text', 'agreeCount', 'disagreeCount', 'id']
tips_filtered = tips_df.loc[:, filtered_columns]

# display user's tips
tips_filtered

text  \
0  The owner is a sexy lady. Call her pretty and she'll give you complimentary wine.     
1  The official theme song is Classy Girls by The Lumineers.                             
2  Always have a cocktail!                                                               
3  They'll make off menu cocktails for you too.  Just tell the waitress what you like!   
4  Try the artichoke bread. A must!                                                      

   agreeCount  disagreeCount                        id  
0  1           0              52bf92e8498e4be29a3ac648  
1  1           0              50f8ac7ae4b03d5962873671  
2  1           0              50ea50e1e4b0d58a5f50ab7e  
3  4           0              50c41f61e4b032fa829c0151  
4  2           0              4f1b610ee4b0e6badb98f542

In [40]:
tip_id = '4f1b610ee4b0e6badb98f542' # tip id

# define URL
url = 'http://api.foursquare.com/v2/tips/{}?client_id={}&client_secret={}&v={}'.format(tip_id, CLIENT_ID, CLIENT_SECRET, VERSION)

# send GET Request and examine results
result = requests.get(url).json()
print(result['response']['tip']['venue']['name'])
print(result['response']['tip']['venue']['location'])

The Finer Things Club
{'isFuzzed': True, 'lat': 40.714660147121606, 'lng': -73.94186034276925, 'cc': 'US', 'city': 'Brooklyn', 'state': 'NY', 'country': 'United States', 'formattedAddress': ['Leonard St.', 'Brooklyn, NY 11211', 'United States']}


In [44]:
# 1. grab prefix of photo
# 2. grab suffix of photo
# 3. concatenate them using the image size  
Image(url='https://igx.4sqi.net/img/user/300x300/484542633_mK2Yum7T_7Tn9fWpndidJsmw2Hof_6T5vJBKCHPLMK5OL-U5ZiJGj51iwBstcpDLYa3Zvhvis.jpg')

Now lets look at the Philadelphia coffee shops in downtown

In [47]:
address = '201 S. Christopher Columbus Blvd., Philadelphia, PA, United States of America'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

39.9453634 -75.14136680714961


In [48]:
search_query = 'coffee'
radius = 500
print(search_query + ' .... OK!')

coffee .... OK!


In [49]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&ll=39.9453634,-75.14136680714961&v=20200701&query=coffee&radius=500&limit=30'

In [50]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5efca4e411145c373d3b23ce'},
 'response': {'venues': [{'id': '4c6adfe9599fbe9a1662913f',
    'name': 'Coffee Shop',
    'location': {'crossStreet': '3rd and chestnut',
     'lat': 39.94704785,
     'lng': -75.14565978333333,
     'labeledLatLngs': [{'label': 'display',
       'lat': 39.94704785,
       'lng': -75.14565978333333}],
     'distance': 411,
     'postalCode': '19106',
     'cc': 'US',
     'city': 'Philadelphia',
     'state': 'PA',
     'country': 'United States',
     'formattedAddress': ['Philadelphia, PA 19106', 'United States']},
    'categories': [],
    'referralId': 'v-1593615413',
    'hasPerk': False},
   {'id': '4be820a9947820a1967cb4db',
    'name': 'Bodhi Coffee',
    'location': {'address': '410 S 2nd St',
     'crossStreet': 'Pine St',
     'lat': 39.94285617341323,
     'lng': -75.14545797458909,
     'labeledLatLngs': [{'label': 'display',
       'lat': 39.94285617341323,
       'lng': -75.14545797458909}],
     'distance

In [51]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

categories  \
0  []                                                                                                                                                                                                                                        
1  [{'id': '4bf58dd8d48988d16d941735', 'name': 'Café', 'pluralName': 'Cafés', 'shortName': 'Café', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/cafe_', 'suffix': '.png'}, 'primary': True}]                              
2  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'Coffee Shop', 'pluralName': 'Coffee Shops', 'shortName': 'Coffee Shop', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_', 'suffix': '.png'}, 'primary': True}]   
3  [{'id': '4d4b7105d754a06374d81259', 'name': 'Food', 'pluralName': 'Food', 'shortName': 'Food', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_', 'suffix': '.png'}, 'primary': True}]                            
4  [{'id': '4bf58dd8d48988d130941735', 'name': 'Building', 'pluralName': 'Buildings', 'shortName': 'Building', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/building/default_', 'suffix': '.png'}, 'primary': True}]           

  delivery.id              delivery.provider.icon.name  \
0  NaN         NaN                                       
1  1914905     /delivery_provider_grubhub_20180129.png   
2  1525855     /delivery_provider_grubhub_20180129.png   
3  NaN         NaN                                       
4  NaN         NaN                                       

              delivery.provider.icon.prefix delivery.provider.icon.sizes  \
0  NaN                                       NaN                           
1  https://fastly.4sqi.net/img/general/cap/  [40, 50]                      
2  https://fastly.4sqi.net/img/general/cap/  [40, 50]                      
3  NaN                                       NaN                           
4  NaN                                       NaN                           

  delivery.provider.name  \
0  NaN                     
1  grubhub                 
2  grubhub                 
3  NaN                     
4  NaN                     

                                                                                                                                                                                                         delivery.url  \
0  NaN                                                                                                                                                                                                                  
1  https://www.grubhub.com/restaurant/bodhi-coffee-410-south-2nd-street-philadelphia/1914905?affiliate=1131&utm_source=foursquare-affiliate-network&utm_medium=affiliate&utm_campaign=1131&utm_content=1914905          
2  https://www.grubhub.com/restaurant/customs-coffee-house-211-chestnut-street-philadelphia/1525855?affiliate=1131&utm_source=foursquare-affiliate-network&utm_medium=affiliate&utm_campaign=1131&utm_content=1525855   
3  NaN                                                                                                                                                                                                                  
4  NaN                                                                                                                                                                                                                  

   hasPerk                        id location.address  ... location.distance  \
0  False    4c6adfe9599fbe9a1662913f  NaN              ...  411                
1  False    4be820a9947820a1967cb4db  410 S 2nd St     ...  447                
2  False    507c17aae4b02bab1b137755  211 Chestnut St  ...  441                
3  False    4f32207419836c91c7b8acc9  108 Chestnut St  ...  326                
4  False    4eb9917af5b994cb99b11ccd  Front Street     ...  464                

                                    

In [52]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
       return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name         categories          address  cc  \
0  Coffee Shop                   None               NaN              US   
1  Bodhi Coffee                  Café               410 S 2nd St     US   
2  Customs Coffee House          Coffee Shop        211 Chestnut St  US   
3  La Tazza D'Art Coffee House   Food               108 Chestnut St  US   
4  London Coffee House           Building           Front Street     US   
5  Stumptown Coffee              Coffee Shop        NaN              US   
6  Xando Coffee and Bar          Coffee Shop        215 Lombard St   US   
7  Chez Ben Bistro & Coffee Bar  French Restaurant  401 Chestnut St  US   

           city        country       crossStreet  distance  \
0  Philadelphia  United States  3rd and chestnut  411        
1  Philadelphia  United States  Pine St           447        
2  Philadelphia  United States  NaN               441        
3  Philadelphia  United States  NaN               326        
4  Philadelphia  United States  Market Street     464        
5  Philadelphia  United States  NaN               484        
6  Philadelphia  United States  NaN               515        
7  Philadelphia  United States  NaN               691        

                                                        formattedAddress  \
0  [Philadelphia, PA 19106, United States]                                 
1  [410 S 2nd St (Pine St), Philadelphia, PA 19147, United States]         
2  [211 Chestnut St, Philadelphia, PA 19106, United States]                
3  [108 Chestnut St, Philadelphia, PA 19106, United States]                
4  [Front Street (Market Street), Philadelphia, PA 19106, United States]   
5  [Philadelphia, PA 19147, United States]                                 
6  [215 Lombard St, Philadelphia, PA 19147, United States]                 
7  [401 Chestnut St, Philadelphia, PA 19106, United States]                

                                                                                                                            labeledLatLngs  \
0  [{'label': 'display', 'lat': 39.94704785, 'lng': -75.14565978333333}]                                                                     
1  [{'label': 'display', 'lat': 39.94285617341323, 'lng': -75.14545797458909}]                                                               
2  [{'label': 'display', 'lat': 39.94847394561634, 'lng': -75.14458164676223}, {'label': 'entrance', 'lat': 39.948344, 'lng': -75.144722}]   
3  [{'label': 'display', 'lat': 39.948036193847656, 'lng': -75.14293670654297}]                                                              
4  [{'label': 'display', 'lat': 39.94950401362793, 'lng': -75.14201088503086}]                                                               
5  [{'label': 'display', 'lat': 39.942624110643294, 'lng': -75.1457791232607}]                                                               
6  [{'label': 'display', 'lat': 39.942514, 'lng': -75.146135}]                                                                               
7  [{'label': 'display', 'lat': 39.948963, 'lng': -75.14797}]                                                                                

         lat        lng      neighborhood postalCode state  \
0  39.947048 -75.145660  NaN               19106      PA     
1  39.942856 -75.145458  NaN               19147      PA     
2  39.948474 -75.144582  NaN               19106      PA     
3  39.948036 -75.142937  NaN               19106      PA     
4  39.949504 -75.142011  NaN               19106      PA     
5  39.942624 -75.145779  NaN               19147      PA     
6  39.942514 -75.146135  NaN               19147      PA     
7  39.948963 -75.147970  Center City East  19106      PA     

                         id  
0  4c6adfe9599fbe9a1662913f  
1  4be820a9947820a1967cb4db  
2  507c17aae4b02bab1b137755  
3  4f32207419836c91c7b8acc9  
4  4eb9917af5b994cb99b11ccd  
5  4eb57b597ee5024d7f09684c  
6  4f32979819836c91c7e6d07d  
7  5b3cf595de3bbf003941952b

In [53]:
dataframe_filtered.name

0    Coffee Shop                 
1    Bodhi Coffee                
2    Customs Coffee House        
3    La Tazza D'Art Coffee House 
4    London Coffee House         
5    Stumptown Coffee            
6    Xando Coffee and Bar        
7    Chez Ben Bistro & Coffee Bar
Name: name, dtype: object

In [54]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) # generate map centred around the hilton hotel

# add a red circle marker to represent the hilton hotel
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Hilton Philadelphia at Penns Landing',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)
# add the Italian restaurants as blue circle markers
for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.categories):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)
    # display map
venues_map

In [55]:
venue_id = '4be820a9947820a1967cb4db' # ID of bodhi coffe
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4be820a9947820a1967cb4db?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [56]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'delivery', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'popular', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4be820a9947820a1967cb4db',
 'name': 'Bodhi Coffee',
 'contact': {'phone': '2672392928',
  'formattedPhone': '(267) 239-2928',
  'twitter': 'bodhicoffee'},
 'location': {'address': '410 S 2nd St',
  'crossStreet': 'Pine St',
  'lat': 39.94285617341323,
  'lng': -75.14545797458909,
  'labeledLatLngs': [{'label': 'display',
    'lat': 39.94285617341323,
    'lng': -75.14545797458909}],
  'postalCode': '19147',
  'cc': 'US',
  'city': 'Philadelphia',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['410 S 2nd St (Pine St)',
   'Philadelphia, PA 19147',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/bodhi-coffee/4be820a9947820a1967cb4db',
 'categories': [{'id': '4bf58dd8d48988d16d941735',
   'name': 'Café',
   'pluralName': 'Cafés',
   'shortName': 'Café',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/cafe_',
    'suffix': '.png'},
   'primary': True},
  {'id': '4bf58dd8d48988d1e0931735',
   'name': 'Coffee Shop',
   'plura

In [57]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

8.6


In [58]:
result['response']['venue']['tips']['count']

23

In [59]:
## Ecco Tips
limit = 15 # set limit to be greater than or equal to the total number of tips
url = 'https://api.foursquare.com/v2/venues/{}/tips?client_id={}&client_secret={}&v={}&limit={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION, limit)

results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5efca6497f80693691002cc2'},
 'response': {'tips': {'count': 23,
   'items': [{'id': '4e7cbe53b3ad0e8263d8949f',
     'createdAt': 1316798035,
     'text': 'At Bodhi Coffee in Society Hill, the café proudly wears salvaged materials, the Stumptown coffees are available in pour-over and drip styles, and the food is fresh, organic and local.',
     'type': 'user',
     'canonicalUrl': 'https://foursquare.com/item/4e7cbe53b3ad0e8263d8949f',
     'lang': 'en',
     'likes': {'count': 2,
      'groups': [{'type': 'others',
        'count': 2,
        'items': [{'id': '16697800',
          'firstName': 'Thirsty',
          'lastName': 'J',
          'photo': {'prefix': 'https://fastly.4sqi.net/img/user/',
           'suffix': '/16697800_0SO6ajOl_2JcveHiPsa8AbCXNznoyev1VQjUJ9LOX-scoJZcs2GYBf0ukRg1WCHsdpChg3TPi.jpg'}},
         {'id': '48618321',
          'firstName': 'Find a Coffice',
          'photo': {'prefix': 'https://fastly.4sqi.net/img/user/',
      

In [60]:
tips = results['response']['tips']['items']

tip = results['response']['tips']['items'][0]
tip.keys()

dict_keys(['id', 'createdAt', 'text', 'type', 'canonicalUrl', 'lang', 'likes', 'logView', 'agreeCount', 'disagreeCount', 'todo', 'user'])

In [61]:
pd.set_option('display.max_colwidth', -1)

tips_df = json_normalize(tips) # json normalize tips

# columns to keep
filtered_columns = ['text', 'agreeCount', 'disagreeCount', 'id', 'user.firstName', 'user.lastName', 'user.gender', 'user.id']
tips_filtered = tips_df.loc[:, filtered_columns]

# display tips
tips_filtered

text  \
0  At Bodhi Coffee in Society Hill, the café proudly wears salvaged materials, the Stumptown coffees are available in pour-over and drip styles, and the food is fresh, organic and local.   

   agreeCount  disagreeCount                        id user.firstName  \
0  3           0              4e7cbe53b3ad0e8263d8949f  Visit Philly    

   user.lastName  user.gender  user.id  
0 NaN            NaN           1666563

In [ ]:
user_id = '1666563' # user ID with most agree counts and complete profile

url = 'https://api.foursquare.com/v2/user/{}?client_id={}&client_secret={}&v={}'.format(user_id, CLIENT_ID, CLIENT_SECRET, VERSION) # define URL

# send GET request
results = requests.get(url).json()
user_data = results['response']['user']

# display features associated with user
user_data.keys()

In [ ]:
print('First Name: ' + user_data['firstName'])
print('Last Name: ' + user_data['lastName'])
print('Home City: ' + user_data['homeCity'])

In [ ]:
user_data['tips']

In [65]:
# define tips URL
url = 'https://api.foursquare.com/v2/users/{}/tips?client_id={}&client_secret={}&v={}&limit={}'.format(user_id, CLIENT_ID, CLIENT_SECRET, VERSION, limit)

# send GET request and get user's tips
results = requests.get(url).json()
tips = results['response']['tips']['items']

# format column width
pd.set_option('display.max_colwidth', -1)
# format column width
pd.set_option('display.max_colwidth', -1)

tips_df = json_normalize(tips)

# filter columns
filtered_columns = ['text', 'agreeCount', 'disagreeCount', 'id']
tips_filtered = tips_df.loc[:, filtered_columns]

# display user's tips
tips_filtered

text  \
0   If you haven't had a chance to try Crime & Punishment's brews yet, keep an eye out for their Pope Francis-related gose "Jesus Wept", a wheat beer brewed with salt and coriander.                          
1   2nd Story Brewing Co. will go way back in time to welcome Pope Francis to Philly. Their beer "Pater Noster" (or "Our Father" in Latin) is usually made by Trappist monks to enjoy on special occasions.    
2   Brewers incorporate ingredients from Pope Francis' home country of Argentina in "Papal Pleasure", a Belgian amber brewed with South American candi sugar and infused with oak from Malbec wine barrels.    
3   Try the special "Pap-Ale" brew. A "session"-able Abbey Dubbel, the beer's surprise ingredient is Yerba Mate, which Pope Francis has said he drinks every day to calm him and to stay healthy.              
4   Philadelphia Brewing Company welcomes Pope Francis in a big way with "Holy Wooder", a limited, draft-only Belgian-style Tripel with a name that plays on the way some Philadelphians pronounce "water."    
5   For the "Divine Dining: A Meal Fit for a Pope" City Food Tour on Thursday, Sept. 24, attendees will work in small groups to create a papal-inspired three-course meal.                                     
6   This no-frills South Philly joint is getting in on the fun on East Passyunk Ave. with an outdoor block party and a DJ on Saturday, Sept. 26 and Sunday, Sept. 27.                                          
7   Celebrating the arrival of their namesake (of sorts), POPE a.k.a. Pub on Passyunk East will host an outdoor block party on Sept. 26-27, featuring food and drink specials, live music, games and more.     
8   Set on 50 floors above the Benjamin Franklin Parkway, Top of the Tower will offer SkyBrunch on Saturday, Sept. 26 & Sunday, Sept. 27 and SkyBuffet during the Festival of Families, Saturday, Sept. 26.    
9   The mural of famous Franks (including Frank Zappa & Benjamin Franklin) now includes Pope Francis. Admire the art, then step into the iconic bar, which draws patrons with cheap beer & a great jukebox.    
10  Inspired by Pope Francis' visit to Philadelphia, Pizza Brain will feature a special-edition Pope pizza box. Locally based Savona-Stavola Foodservice is distributing the custom pizza box to area shops.   
11  In honor of Pope Francis' visit to Philadelphia in September, guests can look forward to special prices on a bevy of Trappist beers, from Chimay to Spencer and more.                                      
12  On the weekend of the papal visit, enjoy the "Châteauneuf-du-Pope" wine promotion with two different Châteauneuf Du Pape wines at the three Tria Cafe locations throughout the city.                       
13  Also known as the P.O.P.E., this cavern-like tavern is a hipster haven. The inclusive menu, ranging from the P.O.P.E.’s namesake burger to vegetarian specialties like the seitan cheesesteak.             
14  Chestnut Hill's Night Kitchen Bakery welcomes Pope Francis with a series of Peaceful Pope shortbread cookies.                                                                                              

    agreeCount  disagreeCount                        id  
0   1           0              55e724cf498eb7ec395c850e  
1   0           0              55e723b8498eb5b1156f05e2  
2   0           0              55e7228b498eb7ec395a6d2a  
3   0           0              55e7170f498eb7ec394ff49d  
4   3           0              55e7167a498eb7ec394f5e2b  
5   0           0              55e71559498eb7ec394e5452  
6   0           0              55e71438498eb7ec394d5617  
7   1           0              55e7141d498eb5b11560be94  
8   1           0              55e71254498e28de2777e9f0  
9   4           0              55e711f3498e28de2777b6af  
10  0           0              55e7113f498eb7ec394accbb  
11  0           1              55e7106d498eb5b1155d9433  
12  0           0              55e71021498eb5b1155d4fa5  
13  1           0              55e70fee498eb5b1155d1c13  
1

In [66]:
tip_id = '55e711f3498e28de2777b6af' # tip id

# define URL
url = 'http://api.foursquare.com/v2/tips/{}?client_id={}&client_secret={}&v={}'.format(tip_id, CLIENT_ID, CLIENT_SECRET, VERSION)

# send GET Request and examine results
result = requests.get(url).json()
print(result['response']['tip']['venue']['name'])
print(result['response']['tip']['venue']['location'])

Dirty Frank's
{'address': '347 S 13th St', 'crossStreet': 'at Pine St', 'lat': 39.945338232462035, 'lng': -75.16284383919128, 'labeledLatLngs': [{'label': 'display', 'lat': 39.945338232462035, 'lng': -75.16284383919128}], 'postalCode': '19107', 'cc': 'US', 'city': 'Philadelphia', 'state': 'PA', 'country': 'United States', 'formattedAddress': ['347 S 13th St (at Pine St)', 'Philadelphia, PA 19107', 'United States']}


In [ ]:
user_friends = json_normalize(user_data['friends']['groups'][0]['items'])
user_friends

In [68]:
# 1. grab prefix of photo
# 2. grab suffix of photo
# 3. concatenate them using the image size  
Image(url='https://igx.4sqi.net/img/user/300x300/484542633_mK2Yum7T_7Tn9fWpndidJsmw2Hof_6T5vJBKCHPLMK5OL-U5ZiJGj51iwBstcpDLYa3Zvhvis.jpg')

In [75]:
venue_id = '4f32979819836c91c7e6d07d' # ID of xando coffe and bar
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4f32979819836c91c7e6d07d?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [76]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'likes', 'dislike', 'ok', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes'])


{'id': '4f32979819836c91c7e6d07d',
 'name': 'Xando Coffee and Bar',
 'contact': {'phone': '2159254910', 'formattedPhone': '(215) 925-4910'},
 'location': {'address': '215 Lombard St',
  'lat': 39.942514,
  'lng': -75.146135,
  'labeledLatLngs': [{'label': 'display',
    'lat': 39.942514,
    'lng': -75.146135}],
  'postalCode': '19147',
  'cc': 'US',
  'city': 'Philadelphia',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['215 Lombard St',
   'Philadelphia, PA 19147',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/xando-coffee-and-bar/4f32979819836c91c7e6d07d',
 'categories': [{'id': '4bf58dd8d48988d1e0931735',
   'name': 'Coffee Shop',
   'pluralName': 'Coffee Shops',
   'shortName': 'Coffee Shop',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_',
    'suffix': '.png'},
   'primary': True}],
 'verified': False,
 'stats': {'tipCount': 0},
 'likes': {'count': 0, 'groups': []},
 'dislike': False,
 'ok': False,
 'allo

In [77]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

This venue has not been rated yet.


In [90]:
venue_id = '4eb57b597ee5024d7f09684c' # ID of 
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4eb57b597ee5024d7f09684c?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [91]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'likes', 'dislike', 'ok', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes'])


{'id': '4eb57b597ee5024d7f09684c',
 'name': 'Stumptown Coffee',
 'contact': {},
 'location': {'lat': 39.942624110643294,
  'lng': -75.1457791232607,
  'labeledLatLngs': [{'label': 'display',
    'lat': 39.942624110643294,
    'lng': -75.1457791232607}],
  'postalCode': '19147',
  'cc': 'US',
  'city': 'Philadelphia',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['Philadelphia, PA 19147', 'United States']},
 'canonicalUrl': 'https://foursquare.com/v/stumptown-coffee/4eb57b597ee5024d7f09684c',
 'categories': [{'id': '4bf58dd8d48988d1e0931735',
   'name': 'Coffee Shop',
   'pluralName': 'Coffee Shops',
   'shortName': 'Coffee Shop',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_',
    'suffix': '.png'},
   'primary': True}],
 'verified': False,
 'stats': {'tipCount': 0},
 'likes': {'count': 1,
  'groups': [{'type': 'others',
    'count': 1,
    'items': [{'id': '26047711',
      'firstName': 'نضال',
      'lastName': 'ا',
      'phot

In [92]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

This venue has not been rated yet.


In [93]:
venue_id = '4eb9917af5b994cb99b11ccd' # ID of london coffee house
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4eb9917af5b994cb99b11ccd?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [94]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'likes', 'dislike', 'ok', 'venueRatingBlacklisted', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto'])


{'id': '4eb9917af5b994cb99b11ccd',
 'name': 'London Coffee House',
 'contact': {},
 'location': {'address': 'Front Street',
  'crossStreet': 'Market Street',
  'lat': 39.94950401362793,
  'lng': -75.14201088503086,
  'labeledLatLngs': [{'label': 'display',
    'lat': 39.94950401362793,
    'lng': -75.14201088503086}],
  'postalCode': '19106',
  'cc': 'US',
  'city': 'Philadelphia',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['Front Street (Market Street)',
   'Philadelphia, PA 19106',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/london-coffee-house/4eb9917af5b994cb99b11ccd',
 'categories': [{'id': '4bf58dd8d48988d130941735',
   'name': 'Building',
   'pluralName': 'Buildings',
   'shortName': 'Building',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/building/default_',
    'suffix': '.png'},
   'primary': True}],
 'verified': False,
 'stats': {'tipCount': 0},
 'likes': {'count': 0, 'groups': []},
 'dislike': False,
 'ok': Fa

In [95]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

This venue has not been rated yet.


In [101]:
venue_id = '507c17aae4b02bab1b137755' # ID of 

url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/507c17aae4b02bab1b137755?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [102]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'delivery', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '507c17aae4b02bab1b137755',
 'name': 'Customs Coffee House',
 'contact': {},
 'location': {'address': '211 Chestnut St',
  'lat': 39.94847394561634,
  'lng': -75.14458164676223,
  'labeledLatLngs': [{'label': 'display',
    'lat': 39.94847394561634,
    'lng': -75.14458164676223},
   {'label': 'entrance', 'lat': 39.948344, 'lng': -75.144722}],
  'postalCode': '19106',
  'cc': 'US',
  'city': 'Philadelphia',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['211 Chestnut St',
   'Philadelphia, PA 19106',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/customs-coffee-house/507c17aae4b02bab1b137755',
 'categories': [{'id': '4bf58dd8d48988d1e0931735',
   'name': 'Coffee Shop',
   'pluralName': 'Coffee Shops',
   'shortName': 'Coffee Shop',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_',
    'suffix': '.png'},
   'primary': True}],
 'verified': False,
 'stats': {'tipCount': 5},
 'likes': {'count': 24,
  'groups': [

In [103]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

6.7


In [96]:
venue_id = '4c6adfe9599fbe9a1662913f' # ID of london coffee house
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4c6adfe9599fbe9a1662913f?client_id=KCPXGKELOGCTZN0DGRRKPXVLK2C5VMZINO2HB5BO4GLFFMK1&client_secret=LX32BLC532BSJZW1HDAIM5KSAGR2TGAEWN4HQCF3NID1FYLS&v=20200701'

In [97]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'likes', 'dislike', 'ok', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes'])


{'id': '4c6adfe9599fbe9a1662913f',
 'name': 'Coffee Shop',
 'contact': {},
 'location': {'crossStreet': '3rd and chestnut',
  'lat': 39.94704785,
  'lng': -75.14565978333333,
  'labeledLatLngs': [{'label': 'display',
    'lat': 39.94704785,
    'lng': -75.14565978333333}],
  'postalCode': '19106',
  'cc': 'US',
  'city': 'Philadelphia',
  'state': 'PA',
  'country': 'United States',
  'formattedAddress': ['Philadelphia, PA 19106', 'United States']},
 'canonicalUrl': 'https://foursquare.com/v/coffee-shop/4c6adfe9599fbe9a1662913f',
 'categories': [],
 'verified': False,
 'stats': {'tipCount': 0},
 'likes': {'count': 0, 'groups': []},
 'dislike': False,
 'ok': False,
 'beenHere': {'count': 0,
  'unconfirmedCount': 0,
  'marked': False,
  'lastCheckinExpiredAt': 0},
 'specials': {'count': 0, 'items': []},
 'photos': {'count': 0, 'groups': []},
 'reasons': {'count': 0, 'items': []},
 'hereNow': {'count': 0, 'summary': 'Nobody here', 'groups': []},
 'createdAt': 1282072553,
 'tips': {'count'

In [98]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

This venue has not been rated yet.


# this concludes my search of the data of the coffee shops downtown in both citys 

My findings were that there were not as many coffee shops in Philadelphia as there were in pittsburgh 7 - 24

Also there were not as many that people had put ratings in for on foursquare

The ratings seem to be the same for the shops in Philadelphia as there were in pittsburgh  8.6, 6.6  / 6.6, 6.2, 8

the tips were helpful in evauating what types of customer service and level of the coffee shops

the users also had comment that will help decide where to start the coffee shop.

Before I could make any conclusioning I would need to gather and analyze more data.